securedrop_export/disk/cli.py

```diff
@@ -0,0 +1,423 @@
+import logging
+import os
+import subprocess
+
+from typing import List, Optional
+
+from securedrop_export.exceptions import ExportException
+
+from .volume import EncryptionScheme, Volume
+from .new_status import Status
+
+logger = logging.getLogger(__name__)
+
+
+class CLI:
+    """
+    A Python wrapper for various shell commands required to detect, map, and
+    mount Export devices.
+
+    CLI callers must handle ExportException and all exceptions and exit with
+    sys.exit(0) so that another program does not attempt to open the submission.
+    """
+
+    # Default mountpoint (unless drive is already mounted manually by the user)
+    _DEFAULT_MOUNTPOINT = "/media/usb"
+
+    def get_connected_devices(self) -> List[str]:
+        """
+        List all block devices attached to VM that are disks and not partitions.
+        Return list of all removable connected block devices.
+
+        Raise ExportException if any commands fail.
+        """
+        logger.info("Checking connected volumes")
+        try:
+            lsblk = subprocess.Popen(
+                ["lsblk", "-o", "NAME,TYPE"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            grep = subprocess.Popen(
+                ["grep", "disk"],
+                stdin=lsblk.stdout,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            command_output = grep.stdout.readlines()
+
+            # The first word in each element of the command_output list is the device name
+            attached_devices = [x.decode("utf8").split()[0] for x in command_output]
+
+        except subprocess.CalledProcessError as ex:
+            raise ExportException(sdstatus=Status.DEVICE_ERROR) from ex
+
+        return self._get_removable_devices(attached_devices)
+
+    def _get_removable_devices(self, attached_devices: List[str]) -> List[str]:
+        """
+        Determine which block devices are USBs by selecting those that are removable.
+        """
+        logger.info("Checking removable devices")
+        usb_devices = []
+        for device in attached_devices:
+            is_removable = False
+            try:
+                removable = subprocess.check_output(
+                    ["cat", f"/sys/class/block/{device}/removable"],
```

Comment:
What is being shown here is the device identifier assigned by the OS. (`/sys/class/block/xvdc/removable`, for example). I do not think sanitization is required.

Reply:
I'm on the fence because that string originates outside of the scope of the function. But I'm not overly concerned by it either.

cc: @l3th3 @lsd-cat in case you've got any advice to give us. (I'd otherwise second @rocodes' suggestion that we keep this as it is, with no sanitization.)